gobblin-data-management/src/main/java/org/apache/gobblin/data/management/retention/DatasetCleaner.java

```diff
@@ -176,7 +176,7 @@ public void onSuccess(Void arg0) {
   @Override
   public void close() throws IOException {
     try {
-      if (this.finishCleanSignal.isPresent()) {
+      if (this.finishCleanSignal != null && this.finishCleanSignal.isPresent()) {
```

Comment:
Yes I found that if an exception is thrown during setup (e.g. issue instantiating the datasetsFinder), this will still be null which will cause the job to fail without actually logging the exception.